libp2p/vbuffer.nim

```diff
@@ -24,9 +24,15 @@ template isEnough*(vb: VBuffer, length: int): bool =
   ## Returns ``true`` if buffer ``vb`` holds at least ``length`` bytes.
   len(vb.buffer) - vb.offset - length >= 0
 
+proc high*(vb: VBuffer): int =
+  ## Returns number of bytes left in buffer ``vb``.
+  result = vb.buffer.high - vb.offset
+  doAssert(result >= 0)
+
 proc len*(vb: VBuffer): int =
   ## Returns number of bytes left in buffer ``vb``.
   result = len(vb.buffer) - vb.offset
+  doAssert(result >= 0)
```

Comment:
Why doAssert is needed here?

Reply:
well the result might be indeed negative since we are doing a subtraction, should be `assert` tho not `doAssert` maybe